### PR TITLE
Update action everytime

### PR DESCRIPTION
**Pull request by srisankethu**
Fixes https://github.com/coala-analyzer/coala/issues/1825
Takes in `failed_actions` and asks  for related params in case of a failed execution

### CODE REVIEW

**Comment by tushar-rishav** [comment]
@srisankethu Seems like you need to:

- Add tests for the changes. That'd probably clear the tests and coverage.
 
- Fix commit. Please follow the commit guidelines.

-  Finally, please run tests locally before sending PR.

**Comment by sudheesh001** [comment]
unack d716662

**Comment by srisankethu** [comment]
@tushar-rishav The PR solves the issue( as far as I have checked). Just need to add tests

**Comment by sils1297** [comment]
hm @srisankethu it's definitely going in the right direction now, I wonder if we could do something simpler to avoid passing down the whole failed actions list everywhere...

**Comment by srisankethu** [comment]
@sils1297 failed_actions moves around 3 functions only ! I think it is simple enough as the problem was in appending section, so we need it in get_action_info and we need to check it in ask_action _and_apply.

**Comment by abhsag24** [comment]
unack  89cd6b4

**Comment by abhsag24** [comment]
some minor changes and a rebase, though i like where this is going :)

**Comment by sils1297** [comment]
unack e563cd3, but we're getting to it indeed

**Comment by sils1297** [comment]
@srisankethu seems test coverage is insufficient

**Comment by tushar-rishav** [comment]
@srisankethu  This is annoying. Would you *please* do these two things before each PR.

- Run `py.test`
- Run `coala`
*Please*?

**Comment by tushar-rishav** [comment]
unack a17a538

**Comment by srisankethu** [comment]
@tushar-rishav I'm sorry if you're feeling annoyed. My tests did a 100% coverage earlier, but @sils1297 pointed out a test case with he thought was unnecessary. So I just removed it, and made a slight change to the other test cases to cover what that test case was covering. It missed an if-condition(...again). So now I have added my previous test case and corrected the test cases according to all the reviews given.

**Comment by AbdealiJK** [comment]
@srisankethu seems like coverage missed again - so marking as wip

**Comment by srisankethu** [comment]
@AbdealiJK Do you any other changes in my PR, it kinda frustating make changes continuously. I hope you understand.

**Comment by AbdealiJK** [comment]
@srisankethu sure - let me do a quick review then

**Comment by srisankethu** [comment]
@AbdealiJK Yes I was thinking about not adding comments to the test because I didn't find any test file having comments.

**Comment by sils1297** [comment]
@srisankethu hey, I can understand that code review can sometimes be frustrating - we know it well, I worked once for months on one PR with @Makman2...

**Comment by srisankethu** [comment]
@sils1297 Wow! Yes, deep down i understand that it's a way to learn and correct yourself :+1:

**Comment by srisankethu** [comment]
Updated changes :)

**Comment by sils1297** [comment]
see old review on https://github.com/coala-analyzer/coala/commit/51c4fe686874931a0ed60edd6e55e699418bae7e

**Comment by sils1297** [comment]
unack https://github.com/srisankethu/coala/commit/b3dc441d621f5f84a1f36805f56e48bfb6e8c582 as per previous review

**Comment by sils1297** [comment]
please read *all* comments from the previous review, correcting some isn't enough and is just more work for both of us.

**Comment by srisankethu** [comment]
@sils1297 Im sorry , I was in a hurry, gotta go for lunch

File: coalib/output/ConsoleInteraction.py [hunk 1]
```diff
@@ -453,13 +455,16 @@ def acquire_settings(log_printer, settings_names_dict):
...
 
-def get_action_info(section, action):
+def get_action_info(section, action, failed_actions):
     """
     Get all the required Settings for an action. It updates the section with
     the Settings.
 
     :param section: The section the action corresponds to.
     :param action:  The action to get the info for.
+    :param failed_actions:  A set of all actions that have failed.A failed
```
Comment (arafsheikh) [inline coalib/output/ConsoleInteraction.py]: Alignment problem. `The action...` should be exactly below `A set...`

Comment (srisankethu) [inline coalib/output/ConsoleInteraction.py]: OK !

**Comment by sils1297** [comment]
tests fail

**Comment by sils1297** [comment]
unack https://github.com/srisankethu/coala/commit/db299ef8c986ae68d5b3b3999e976da63b293f47

**Comment by srisankethu** [comment]
Done :+1:

**Comment by sils1297** [comment]
unack e519580f271f5b90495658ebcb1ac7c0a34569b7

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**